Linkedin/cypress/support/commands.ts

```diff
@@ -0,0 +1,72 @@
+/// <reference types="cypress" />
+// ***********************************************
+// This example commands.ts shows you how to
+// create various custom commands and overwrite
+// existing commands.
+//
+// For more comprehensive examples of custom
+// commands please read more here:
+// https://on.cypress.io/custom-commands
+// ***********************************************
+//
+//
+// -- This is a parent command --
+// Cypress.Commands.add("login", (email, password) => { ... });
+//
+//
+// -- This is a child command --
+// Cypress.Commands.add('drag', { prevSubject: 'element'}, (subject, options) => { ... })
+//
+//
+// -- This is a dual command --
+// Cypress.Commands.add('dismiss', { prevSubject: 'optional'}, (subject, options) => { ... })
+//
+//
+// -- This will overwrite an existing command --
+// Cypress.Commands.overwrite('visit', (originalFn, url, options) => { ... })
+//
+export { };
+declare global {
+  namespace Cypress {
+    interface Chainable {
+      register(
+        firstName: string,
+        lastName: string,
+        email: string,
+        password: string,
+      ): Chainable<void>;
+      login(email: string, password: string): Chainable<void>;
+      logout(): Chainable<void>;
+    }
+  }
+}
+
+function register(
+  firstName: string,
+  lastName: string,
+  email: string,
+  password: string,
+) {
+  cy.visit("/signup");
+  cy.get("ion-input[label='First Name']").type(firstName);
+  cy.get("ion-input[label='Last Name']").type(lastName);
+  cy.get("ion-input[label='Email or phone number']").type(email);
+  cy.get("ion-input[label='Password (6+ characters)']").type(password);
+  cy.get("ion-button.agree-join-btn").click();
```

Comment:
<div>

<strong>Bito Code Review Agent Run</strong> <a href=https://github.com/omkardongre/Linkedin-Clone/pull/13#issuecomment-2392147368>#d155d6</a><span> - 10/03/2024, 07:23 pm</span>
<br>

<div><small>🔴</small> High importance</div>

<div id="suggestion">
<div id="issue">
<b>Issue:</b> The selectors used in the register function are based on labels, which might be fragile and prone to breaking if the UI changes.
</div>
<div id="fix">
<b>Fix:</b> Replace label-based selectors with data-* attributes. Add these attributes to the corresponding elements in your application code.
</div>
<details>
<summary>
<b>Code suggestion</b>
</summary>
<div id="code">

```
 @@ -51,5 +51,5 @@
 -  cy.get("ion-input[label='First Name']").type(firstName);
 -  cy.get("ion-input[label='Last Name']").type(lastName);
 -  cy.get("ion-input[label='Email or phone number']").type(email);
 -  cy.get("ion-input[label='Password (6+ characters)']").type(password);
 -  cy.get("ion-button.agree-join-btn").click();
 +  cy.get("[data-cy=first-name-input]").type(firstName);
 +  cy.get("[data-cy=last-name-input]").type(lastName);
 +  cy.get("[data-cy=email-input]").type(email);
 +  cy.get("[data-cy=password-input]").type(password);
 +  cy.get("[data-cy=signup-button]").click();
```
</div>
</details>
</div>
<br>


</div>

Is this a valid issue, or was it incorrectly flagged by the Agent?
- [ ] it was incorrectly flagged

Reply:
good point raised